[CI] MixedClusterClientYamlTestSuiteIT test {p0=indices.create/20_synthetic_source/object with unmapped fields} failing

**Build Scans:**
- [elasticsearch-intake #12141 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/hujov7mjsz246)
- [elasticsearch-pull-request #39042 / 8.15.4_bwc](https://gradle-enterprise.elastic.co/s/cg32yvgnh35bk)
- [elasticsearch-pull-request #39042 / 8.15.4_bwc](https://gradle-enterprise.elastic.co/s/komo6dgeretyk)
- [elasticsearch-intake #12132 / 8.15.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4r5b7jtlwtoa6)

**Reproduction Line:**
```
./gradlew ":qa:mixed-cluster:v8.15.4#mixedClusterTest" -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=indices.create/20_synthetic_source/empty object with unmapped fields}" -Dtests.seed=5B2AD850B8CD6E7D -Dtests.bwc=true -Dtests.locale=mi-Latn-NZ -Dtests.timezone=Etc/Greenwich -Druntime.java=23
```

**Applicable branches:**
8.16

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dindices.create%2F20_synthetic_source%2Fobject%20with%20unmapped%20fields%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [indices.create/20_synthetic_source:194]: field [hits.hits.0._source.path.some.other\.name] is null
```

**Issue Reasons:**
- [8.16] 2 consecutive failures in step 8.15.4_bwc
- [8.16] 2 consecutive failures in pipeline elasticsearch-intake
- [8.16] 4 failures in test test {p0=indices.create/20_synthetic_source/object with unmapped fields} (11.8% fail rate in 34 executions)
- [8.16] 2 failures in step 8.15.4_bwc-snapshots (11.8% fail rate in 17 executions)
- [8.16] 2 failures in step 8.15.4_bwc (40.0% fail rate in 5 executions)
- [8.16] 2 failures in pipeline elasticsearch-intake (22.2% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -340,15 +340,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.InferenceCrudIT
   method: testSupportedStream
   issue: https://github.com/elastic/elasticsearch/issues/113430
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=indices.create/20_synthetic_source/object with unmapped fields}
-  issue: https://github.com/elastic/elasticsearch/issues/115910
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=indices.create/20_synthetic_source/empty object with unmapped fields}
-  issue: https://github.com/elastic/elasticsearch/issues/115911
-- class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
-  method: test {p0=indices.create/20_synthetic_source/nested object with unmapped fields}
-  issue: https://github.com/elastic/elasticsearch/issues/115912
 - class: org.elasticsearch.xpack.spatial.search.GeoGridAggAndQueryConsistencyIT
   method: testGeoShapeGeoHash
   issue: https://github.com/elastic/elasticsearch/issues/115664

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.create/20_synthetic_source.yml
@@ -1,6 +1,6 @@
 object with unmapped fields:
   - requires:
-      cluster_features: ["mapper.track_ignored_source"]
+      cluster_features: ["mapper.ignored_source.dont_expand_dots"]
       reason: requires tracking ignored source
 
   - do:
@@ -100,7 +100,7 @@ unmapped arrays:
 ---
 nested object with unmapped fields:
   - requires:
-      cluster_features: ["mapper.track_ignored_source"]
+      cluster_features: ["mapper.ignored_source.dont_expand_dots"]
       reason: requires tracking ignored source
 
   - do:
@@ -152,7 +152,7 @@ nested object with unmapped fields:
 ---
 empty object with unmapped fields:
   - requires:
-      cluster_features: ["mapper.track_ignored_source"]
+      cluster_features: ["mapper.ignored_source.dont_expand_dots"]
       reason: requires tracking ignored source
 
   - do:

